pkg/apis/hive/v1/clusterclaim_types.go

```diff
@@ -15,24 +15,37 @@ type ClusterClaimSpec struct {
 	// +optional
 	Subjects []rbacv1.Subject `json:"subjects,omitempty"`
 
-	// Namespace is the namespace containing the ClusterDeployment of the claimed cluster.
-	// This field will be set by the ClusterPool when the claim is assigned a cluster.
-	// +optional
-	Namespace string `json:"namespace,omitempty"`
-
 	// Lifetime is the maximum lifetime of the claim after it is assigned a cluster. If the claim still exists
 	// when the lifetime has elapsed, the claim will be deleted by Hive.
 	// +optional
 	Lifetime *metav1.Duration `json:"lifetime,omitempty"`
+
+	// Namespace is the namespace containing the ClusterDeployment of the claimed cluster.
+	// This field will be set by the ClusterPool when the claim is assigned a cluster, and the cluster has been
+	// successfully resumed from hibernation. Callers should monitor this field to know when the cluster is ready
+	// for use.
+	// The ClusterDeployment name will match the Namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
 }
 
 // ClusterClaimStatus defines the observed state of ClusterClaim.
 type ClusterClaimStatus struct {
 	// Conditions includes more detailed status for the cluster pool.
 	// +optional
 	Conditions []ClusterClaimCondition `json:"conditions,omitempty"`
+
+	// Namespace is the namespace containing the ClusterDeployment of the claimed cluster.
+	// This field will be set by the ClusterPool when a claim is assigned to a cluster.
+	// Even when set the ClusterDeployment may not yet have resumed from hibernation, for callers who would prefer to wait
+	// for this the spec.namespace field will only be assigned once the cluster is running.
+	// The ClusterDeployment name will match the Namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
I am very tempted to drop this and just go back to only Spec.Namespace. I don't like that there are two fields of the same name and no strong indication what they both mean. 

The condition I added later is feeling like enough, you can oc wait on it until it's ready and then proceed.